src/main/java/org/cryptomator/cryptofs/health/type/AmbiguousType.java

```diff
@@ -0,0 +1,42 @@
+package org.cryptomator.cryptofs.health.type;
+
+import org.cryptomator.cryptofs.common.CiphertextFileType;
+import org.cryptomator.cryptofs.health.api.CommonDetailKeys;
+import org.cryptomator.cryptofs.health.api.DiagnosticResult;
+
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A ciphertext dir ending with c9r or c9s but does contain more than one valid signature file.
+ * <p>
+ * Valid signature files for c9r are {@value org.cryptomator.cryptofs.common.Constants#DIR_FILE_NAME} and {@value org.cryptomator.cryptofs.common.Constants#SYMLINK_FILE_NAME}.
+ * Valid signature files for c9s are the ones for c9r and {@value org.cryptomator.cryptofs.common.Constants#CONTENTS_FILE_NAME}.
+ */
+public class AmbiguousType implements DiagnosticResult {
```

Comment:
to draft a fix: Create sibling nodes with "_2", "_3", for each possible type and move one type-specific files to them

Reply:
Feasible, but it requires to build up the decrypted dir structure up to directory containing this ambiguous type because we need the dirId to get/create a cleartext filename. In the worst case the fix fails because we are inside an orphaned directory, for which no dirId exists anymore.

Actually, i think, for several checks knowing this structure would be benefical (e.g. providing decrypted names in `DiagnosticResult::details`), and hence suggest to build it in the first check requiring it and then caching it for later checks (done with lazy dagger injection?). Of course this would need a not too small amount of code to write and test and is out of scope for this PR.